src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService.cs

```diff
@@ -149,8 +150,11 @@ public async Task ForceAnalyzeAsync(Solution solution, ProjectId? projectId = nu
                     var index = 0;
                     foreach (var project in solution.Projects)
                     {
-                        tasks[index++] = Task.Run(
-                            () => analyzer.ForceAnalyzeProjectAsync(project, cancellationToken));
+                        tasks[index++] = Task.Run(async () =>
+                            {
+                                await analyzer.ForceAnalyzeProjectAsync(project, cancellationToken).ConfigureAwait(false);
+                                onProjectAnalyzed(project);
```

Comment:
if things get canceled is it ok if we don't call this?  i'm always very wary of these sorts of systems because i don't know how status interleaves with progress, and if they should be distinct, or should have invariants between the two.

Reply:
Yes, I can adjust the caller to explicitly handle the case where all projects did not complete analysis.